### ISSUE TITLE
Custom request body for HTTP sink

### ISSUE THREAD

**Issue by Anonymodesu** [vectordotdev/vector#15963]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

Using the [http sink](https://vector.dev/docs/reference/configuration/sinks/http/), I'd like to send database query statements to the [Neo4j HTTP API](https://neo4j.com/docs/http-api/current/actions/begin-and-commit-a-transaction-in-one-request/). Neo4j expects the request body to look like:

```json
{
  "statements": [
    {
      "statement": "<query statement 1>",
      "parameters": "<params 1>"
    },
    {
      "statement": "<query statement 2>",
      "parameters": "<params 2>"
    },
    ...
  ]
}
```

However, the http sink wraps the logs from [`inputs`](https://vector.dev/docs/reference/configuration/sinks/http/) into an array like so:
```json
[
    {
        "statements": [
            {
                "statement": "<query statement 1>",
                "parameters": "<params 1>"
            }
        ]
    },
    {
        "statements": [
            {
                "statement": "<query statement 2>",
                "parameters": "<params 2>"
            }
        ]
    },
    ...
]
```
Can we add a configuration option that lets us modify this request body?

### Attempted Solutions

Currently I'm constructing the request body like so:
```toml
[transforms.wrap_neo4j_syslog]
inputs = ["parse_syslog"]
type = "remap"
source = '''
  message = to_string!(.message)
  . = {
    "statements": [
      {
        "statement": "CREATE (n:Foobar) SET n.message = $$message RETURN n",
        "parameters": {
          "message": "{{message}}"
        }
      }
    ]
  }
'''
```

### Proposal

Maybe we can specify a root field to batch logs under?

```toml
[sinks.my_sink_id]
type = "http"
batch.field = "statements"
```

### Version

0.26.0

**Comment by jdiebold** [comment]
You could try the Reduce Transform (https://vector.dev/docs/reference/configuration/transforms/reduce/). 
Also maybe this helps #15651. I implemented it for a similar use case (should be available via the nightly build). This way you can make use of the batching capabilities of the http sink. The reduce transform doesn't have these batching configurations yet.

**Comment by Anonymodesu** [comment]
Thanks @jdiebold your feature https://github.com/vectordotdev/vector/pull/15696 works like a charm:

```toml
payload_prefix = '{"statements":'
payload_suffix = "}"
```

Looking forward to it in a stable release :)

I'll close this issue.

**Comment by jdiebold** [comment]
i just saw, it should be included in release 0.28
